codec/yaml.go

```diff
@@ -0,0 +1,32 @@
+package codec
+
+import (
+	"encoding/json"
+
+	"github.com/gogo/protobuf/proto"
+	"gopkg.in/yaml.v2"
+)
+
+// MarshalYAML marshals the provided toPrint content with the provided JSON marshaler
+// by encoding JSON, decoding JSON, and then encoding YAML.
+func MarshalYAML(jsonMarshaler JSONMarshaler, toPrint proto.Message) ([]byte, error) {
```

Comment:
added additional comments there.

Reply:
@clevinson  - you mean that we want to hide some fields? Or because we are using some custom logic to for fields to make proper `[]byte` output (eg things related to keys)?

Reply:
@robert-zaremba eg in case we need to `UnpackInterfaces`, for instance for multisigs.